src/snowflake/snowpark/lineage.py

```diff
@@ -0,0 +1,454 @@
+            depth (int): The depth of the trace, defaults to 2.
+
+        Returns:
+            snowflake.snowpark.DataFrame: A DataFrame representing the traced lineage.
+
+            Example:
+                >>> db = "YOUR_DATABASE"
+                >>> schema = "YOUR_SCHEMA"
+                # Creating a table and views in the session
+                >>> session.sql(f"CREATE OR REPLACE TABLE {db}.{schema}.T1(C1 INT)").collect()
+                >>> session.sql(
+                ...     f"CREATE OR REPLACE VIEW {db}.{schema}.V1 AS SELECT * FROM {db}.{schema}.T1"
+                ... ).collect()
+                >>> session.sql(
+                ...     f"CREATE OR REPLACE VIEW {db}.{schema}.V2 AS SELECT * FROM {db}.{schema}.V1"
+                ... ).collect()
+                # Tracing the lineage of the table
+                >>> df = session.lineage.trace(
+                ...     f"{db}.{schema}.T1",
+                ...     "table",
+                ...     direction=LineageDirection.DOWNSTREAM
+                ... )
+                >>> df.show()
+                ---------------------------------------------------------------------------------------------------------------------------------------
+                | "SOURCE_OBJECT"                                        | "TARGET_OBJECT"                                    | "LINEAGE"   | "DEPTH" |
+                ---------------------------------------------------------------------------------------------------------------------------------------
+                | {"createdOn": "1712881232230", "domain": "TABLE",      | {"createdOn": "1712881232230", "domain": "VIEW",   | "Downstream"| 1       |
+                |  "name": "YOUR_DATABASE.YOUR_SCHEMA.T1", "status":     |  "name": "YOUR_DATABASE.YOUR_SCHEMA.V1", "status": |             |         |
+                |  "ACTIVE"}                                             |  "ACTIVE"}                                         |             |         |
+                | {"createdOn": "1712881232230", "domain": "VIEW",       | {"createdOn": "1712881232230", "domain": "VIEW",   | "Downstream"| 2       |
+                |  "name": "YOUR_DATABASE.YOUR_SCHEMA.V1", "status":     |  "name": "YOUR_DATABASE.YOUR_SCHEMA.V2", "status": |             |         |
+                |  "ACTIVE"}                                             |  "ACTIVE"}                                         |             |         |
+                ---------------------------------------------------------------------------------------------------------------------------------------
+                <BLANKLINE>
+        """
+        if not object_name:
+            raise ValueError("Object name must be provided.")
+        if not object_domain:
+            raise ValueError("Object type must be provided.")
+
+        if depth < 1 or depth > 5:
+            raise ValueError("Depth must be between 1 and 5.")
+
+        if direction == LineageDirection.BOTH and depth == 1:
```

Comment:
is there a reason we need to handle this specially?

Reply:
Its a small optimization. We can get the information in 1 call instead f 2 calls.